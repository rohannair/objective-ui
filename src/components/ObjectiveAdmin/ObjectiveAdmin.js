import React, { PropTypes } from 'react'
import styled from 'styled-components'

import ToggleSwitch from '../Forms/ToggleSwitch'

const ObjectiveAdmin = styled(({
  isOwner,
  objective,
  onPrivateChange,
  className
}) => {
  if (!isOwner) return <div />

  return (
    <div className={className}>
      <ToggleSwitch
        isChecked={objective.isPrivate}
        label="Private to company"
        onChange={onPrivateChange}
      />
    </div>
  )
})`
  display: flex;
  flex-direction: row;
  align-items: center;

  background-color: #fff;
  width: 100%;
  border-top: 1px solid #efefef;
  padding: 10px 30px;
  font-size: 0.875rem;
`

export default ObjectiveAdmin
