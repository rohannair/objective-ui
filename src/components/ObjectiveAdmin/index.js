import ObjectiveAdmin from './ObjectiveAdmin'
export default ObjectiveAdmin
