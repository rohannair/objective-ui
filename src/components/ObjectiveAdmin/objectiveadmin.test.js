import React from 'react'

import ObjectiveAdmin from './index.js'

describe('Test suite for ObjectiveAdmin component', () => {
  it('ObjectiveAdmin should exist', () => {
    let wrapper = shallow(<ObjectiveAdmin/>)
    expect(wrapper).toMatchSnapshot()
  })
})

