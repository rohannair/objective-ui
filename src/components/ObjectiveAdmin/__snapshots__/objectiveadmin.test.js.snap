exports[`Test suite for ObjectiveAdmin component ObjectiveAdmin should exist 1`] = `
<Component
  className="dtcyiV" />
`;
